### DISCUSSION TITLE
Rephrasing introductory text under Design Patterns

### DISCUSSION THREAD

**Discussion by MarkoM-dot** [rust-unofficial/patterns#297] [Ideas]
Hi! I would like to contribute by submitting some changes to the introduction under the heading "Design Patterns". Currently I have:
"In software development, we often come across problems that share similarities 
regardless of the environment they appear in. Although the implementation details
are crucial to solve the task at hand, we may abstract from these particularities to find
some nugget of valuable practices that have a more general application.
Design patterns are a collection of reusable tried and tested solutions to recurring 
problems in engineering. They make our software more modular, maintainable, and extensible.
Moreover, these patterns provide a common language for developers to use, making them 
an excellent tool for effective communication when problem-solving in teams."

This would replace the following:
"When developing programs, we have to solve many problems. A program can be viewed as a solution to a problem. It can also be viewed as a collection of solutions to many problems. All of these solutions work together to solve a bigger problem."

I would like to submit this as [wip] and am open to feedback. Please let me know if you would be interested in changes like these! I would be so happy to contribute!

**Comment by marcoieni** [comment]
Sounds good, thanks! Can you submit a PR?

**Reply by MarkoM-dot** [reply]
Hi thank you for the quick reply!
https://github.com/rust-unofficial/patterns/pull/298
